server/player/player.go

```diff
@@ -1480,6 +1501,10 @@ func (p *Player) handleUseContext(ctx *item.UseContext) {
 	p.SetHeldItems(p.subtractItem(p.damageItem(i, ctx.Damage), ctx.CountSub), left)
 	p.addNewItem(ctx)
 	for _, it := range ctx.ConsumedItems {
+		err := p.offHand.RemoveItem(it)
```

Comment:
There is potential case here where this may remove some but not all of an item, but then attempts with the same amount in the actual inventory,

e.g. if a use costs 2 arrows, and you have 1 arrow in the offhand + 2 arrows in your inventory, this will take the 1 arrow in the offhand and 2 from the inventory